src/test/java/picard/fingerprint/FingerprintCheckerTest.java

```diff
@@ -316,8 +320,45 @@ public void testQueryable(final File vcf, boolean expectedQueryable) {
         }
     }
 
+    @DataProvider
+    Object[][] deepDataProvider() {
+        return new Object[][]{{10}, {50}, {100}, {500}, {1000}, {5000}, {10000}, {50000}};
+    }
+
+    @Test(dataProvider = "deepDataProvider")
+    void testCanHandleDeepData(int counts) {
+        final HaplotypeProbabilitiesFromSequence hp1 = new HaplotypeProbabilitiesFromSequence(hb);
+        addObservation(hp1, hb, counts, hb.getFirstSnp().getAllele1());
+        addObservation(hp1, hb, counts, hb.getFirstSnp().getAllele2());
+
+        final double[] logLikelihoods = hp1.getLogLikelihoods();
+        Assert.assertTrue( MathUtil.min(logLikelihoods) < 0 );
+    }
+
+    @Test(dataProvider = "deepDataProvider")
+    void testCanHandleDeepDataHPFromContamination(int counts) throws IOException {
+        final double maf = 0.4;
+        final Snp snp = new Snp("test", "chr1", 1, (byte) 'A', (byte) 'T', maf, Collections.singletonList("dummy"));
```

Comment:
What about changing the class field snp so that one of the alleles matches the reference, so you can then use the shared `hb`, instead of making a nearly identical local version here?

Reply:
:+1: